### DISCUSSION TITLE
Memory Slots and Downloading Onto Hub

### DISCUSSION THREAD

**Discussion by ns280** [pybricks/support#1216] [General]
Hello,
I am a student part of an FLL team and have a few questions.

1. Can somebody provide code on how to replicate the memory slot system as seen in block programming with Spike? 
2. How do you download programs onto the spike prime hub?
Thanks.

**Comment by laurensvalk** [comment]
We have a new feature to support this. It's not been added to the documentation yet, but you can try this example: https://github.com/pybricks/pybricks-api/issues/144 if you use beta.pybricks.com

**Comment by ns280** [comment]
Thanks for the quick response. Another question about this: Is there a way to preload the next portion of code after on portion is done in memory slots?

**Reply by laurensvalk** [reply]
You can use the same technique to run functions from multiple files after one another.

**Reply by ns280** [reply]
Thanks!
